test/cpp/tensorexpr/test_loopnest.cpp

```diff
@@ -6976,5 +6976,75 @@ TEST(LoopNest, compressBufferIndicesMixed) {
   ASSERT_EQ(dynamic_cast<const IntImm*>(A->dim(1))->value(), 200);
 }
 
+TEST(LoopNest, compressMultipleBuffers) {
+  KernelScope kernel_scope;
+
+  // Input IR:
+  // for (int i = 0; i < 100; ++i) {
+  //   for (int j = 0; j < 200; ++j) {
+  //     A[i,j] = sin(i*j)
+  //   }
+  //   for (int k = 0; k < 199; ++k) {
+  //     B[i,k] = A[i,k] + A[i, k+1]
+  //   }
+  //   for (int m = 0; m < 50; ++m) {
+  //     C[i,m] = B[i,m]
+  //   }
+  // }
+  Buf* A = new Buf("A", {new IntImm(100), new IntImm(200)}, kInt);
+  Buf* B = new Buf("B", {new IntImm(100), new IntImm(200)}, kInt);
+  Buf* C = new Buf("C", {new IntImm(100), new IntImm(200)}, kInt);
+  BufHandle a_buf(A);
+  BufHandle b_buf(B);
+  BufHandle c_buf(C);
```

Comment:
We probably can create just `BufHandle`s - it seems that we use raw buffers only in the final checks, but that can be done through handles as well.

Reply:
Ok, sure. Since there are tests above that does the same, I will do this as a follow up.